From 1e0488878cc75ccd7d8695fc4f7f4c32f6320195 Mon Sep 17 00:00:00 2001
From: Sai Kumar Sanagavarapu <ssanagav@codeaurora.org>
Date: Tue, 8 Aug 2017 16:27:19 +0530
Subject: [PATCH 5/5] camera: Skip HFR checks for privileged apps.

Issue: HAL3 supports HFR only if its greater than or equal
       to 120fps.

Fix: Skip the checks to allow HFR 60 and 90 fps in HAL3 for
     specific apps.
     The following property need to be set to allow these
     HFR configurations :
     adb shell setprop persist.camera.privapp.list <pack1,pack2,etc>

Change-Id: I5f3bc94bea60dbe03284de39cd4280f67df8b015
CRs-Fixed: 2258892
---
 .../hardware/camera2/utils/SurfaceUtils.java  | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/core/java/android/hardware/camera2/utils/SurfaceUtils.java b/core/java/android/hardware/camera2/utils/SurfaceUtils.java
index 6c83057fdf29..65de7060ad19 100644
--- a/core/java/android/hardware/camera2/utils/SurfaceUtils.java
+++ b/core/java/android/hardware/camera2/utils/SurfaceUtils.java
@@ -34,6 +34,11 @@ import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
 
+import android.app.ActivityThread;
+import android.os.SystemProperties;
+import android.text.TextUtils;
+
+
 /**
  * Various Surface utilities.
  */
@@ -241,7 +246,13 @@ public class SurfaceUtils {
                     + " the size must be 1 or 2");
         }
 
+        if (isPrivilegedApp()) {
+            //skip checks for privileged apps
+            return;
+        }
+
         List<Size> highSpeedSizes = null;
+
         if (fpsRange == null) {
             highSpeedSizes = Arrays.asList(config.getHighSpeedVideoSizes());
         } else {
@@ -293,6 +304,23 @@ public class SurfaceUtils {
         }
     }
 
+    private static boolean isPrivilegedApp() {
+        String packageName = ActivityThread.currentOpPackageName();
+        String packageList = SystemProperties.get("persist.vendor.camera.privapp.list");
+
+        if (packageList.length() > 0) {
+            TextUtils.StringSplitter splitter = new TextUtils.SimpleStringSplitter(',');
+            splitter.setString(packageList);
+            for (String str : splitter) {
+                if (packageName.equals(str)) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
     private static native int nativeDetectSurfaceType(Surface surface);
 
     private static native int nativeDetectSurfaceDataspace(Surface surface);
-- 
2.25.1

